src/probes/isWeakCrypto.js

```diff
@@ -0,0 +1,26 @@
+// Internal Dependencies
+import { warnings } from "../constants.js";
+
+function validateNode(node) {
+  const isCallExpression = node.type === "CallExpression";
+  const isSimpleIdentifier = isCallExpression &&
+    node.callee.type === "Identifier" &&
+    node.callee.name === "createHash";
+  const isMemberExpression = isCallExpression &&
+    node.callee.type === "MemberExpression" &&
+    node.callee.property.name === "createHash";
+
+  return [isSimpleIdentifier || isMemberExpression];
+}
+
+function main(node, { analysis }) {
+  const arg = node.arguments.at(0);
+  if (arg.value === "md5" && analysis.dependencies.has("crypto")) {
+    analysis.addWarning(warnings.weakCrypto, "md5", node.loc);
+  }
```

Comment:
Make the code dynamic to check multiple hash algorithms. The list of vulnerable hashes could be stored in an ES6 Set.